script.js

```diff
@@ -1 +1,123 @@
+const form = document.querySelector(".input-box"),
+    input=form.querySelector("input"),
+    waitList=document.querySelector(".wait-list"),
+    completeList=document.querySelector(".complete-list");
 
+const TODO_LS='toDos';
+
+
+
+function loadToDos(){
+    const loadedToDos=localStorage.getItem(TODO_LS);
+    if(loadedToDos!==null){
+        const parsedToDos=JSON.parse(loadedToDos);
+        parsedToDos.forEach(function(todo){
+            paintToDo(todo.text);
+        });
+    }
+}
+
+function saveToDos(){
+    localStorage.setItem(TODO_LS,JSON.stringify(toDos));
+}
+const toDos=[];
+function paintToDo(text){
+    const plusLi = document.createElement('li');
+    const delBtn = document.createElement('img');
+
+    //const img=document.createElement("img");
+    delBtn.src="img/bin.png";
+    //delBtn.innerHTML="$";
+
+    //delBtn.append(img);
+    delBtn.addEventListener("click", deleteToDo);
+    const newId=toDos.length+1;
+    //plusLi.innerHTML = input.value;
+    const inputText=document.createTextNode(input.value);
+    plusLi.id=newId;
+
+    const span = document.createElement("span");
+    span.addEventListener("click", toggleToDo);
+    //span.innerText=text;
+    span.appendChild(inputText);
+    plusLi.appendChild(span);
+    plusLi.appendChild(delBtn);
+    delBtn.hidden=true;
+    plusLi.addEventListener("mouseover",function(){
+        plusLi.childNodes[1].hidden=false;
+    });
+
+    waitList.appendChild(plusLi);
+    countTodos();
+
+    const toDoObj = {
+        text: text,
+        id: newId
+    };
+    //saveToDos();
+    toDos.push(toDoObj);
+}
+
+function toggleToDo(event){
+    let span = event.target;
+    let li = span.parentNode;
+    //if(waitList.has)
+    // waitList.removeChild(li);
+    span.removeEventListener("click", toggleToDo);
+    let completedLi = li.cloneNode(true);
+    li.remove();
+    // completeList.appendChild(li);
+    
+    completedLi.childNodes[0].addEventListener("click", toggleUp);
+    completedLi.childNodes[1].addEventListener("click", deleteToDo);
+    completedLi.childNodes[1].hidden=true;
+    completedLi.addEventListener("mouseover",function(){
+        completedLi.childNodes[1].hidden=false;
+    });
+    completeList.appendChild(completedLi);
+    countTodos();
+}
+
+function toggleUp(event){
```

Comment:
함수의 이름이 모호하기도 하고, 함수의 역할과 이름이 다른 것 같아요.

1. 네이밍
함수의 이름에서는 `toggleUp()`이 어떤 것을 어디로 어떻게 토글하는지에 대해서 구체적으로 드러나있지 않죠?

```suggestion
function toggleCompleteState(todoItem)
```

이런식으로 구체적으로 함수 이름을 지어주시는게 좋습니다.

2. 역할과 이름의 불일치
`toggleUp()`의 뜻을 "todo item의 complete 상태를 바꾼다"라는 의미에서 이름을 지으신 것이라면,
실제로 이 함수는 event handler 역할 또한 같이 하고 있죠?

그런 의미에서 이런식으로 바꿔보는 건 어떨까요?
```javascript
function handleTodoItemClick(clickEvent) {
    const todoItem = getTodoItem(clickEvent); // somehow get todo item object
    toggleCompleteState(todoItem);
}
```

이런 식으로 한번 event handler를 거쳐서 호출하는 방식이 더 의미론적으로 좋아 보여용!